### PR TITLE
Create basic backbone models.

### PR DESCRIPTION
**Pull request by amgrubb**
This pull request creates basic Backbone Models for:
 *  Actor
 *  Link
 *  UserEvaluation
 *  Constraint
 *  FunctionSegment
 *  EvolvingFunction
 *  Intention

### CODE REVIEW

**Comment by amgrubb** [comment]
@leshghi Please review my syntax and what I've created.

File: leaf-ui/rappid-extensions/BackboneModelsIntention.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        repAbsTime: null,       // 0, 1, ..n
+       
+    },
+    
+    //TODO: I think could have some helper functions for the constructor. 
+});
+
+/** This file contains backbone model representations of the original model objects - WIP */
+var IntentionBBM = Backbone.Model.extend({
+    idAttribute: "uid",
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsIntention.js]: Can remove

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        this.type = options.type;               // Options are '=', '<', and '<='
+        this.srcID = options.srcID;
+        this.destID = options.destID;
+        this.srcRefTP = options.srcRefTP;       // Reference Time Point A,B,C, etc.
+        this.destRefTP = options.destRefTP;     // Reference Time Point A,B,C, etc.
+        /* Absolute time points are only used with the '=' type of operator.
+         *   If a timepoint is not given -1 should be assigned as the default value.
+         * */   
+        if (options.absTP == 'undefined') {
+            if (this.type  == '<' || this.type == '<='){
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js:63]: Do we want to delete this nested if/else for the type

File: leaf-ui/rappid-extensions/BackboneModelsIntention.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/** 
+ * This file contains Backbone Models for:
+ *  FunctionSegment
+ *  EvolvingFunction
+ *  Intention
+ */
+
+/** 
+ * Backbone Model of An Atomic Function Segment
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsIntention.js]:
```suggestion
 * 
 */
```

File: leaf-ui/rappid-extensions/BackboneModelsIntention.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        this.refEvidencePair = options.refEvidencePair;   //a.k.a. Evaluation Value
+        this.startTP = options.startTP;
+        this.stopTP = options.stopTP;
+    }
+});
+
+/** 
+ * Backbone Model of Evolving Functions
+ *      TODO: Finish migration from modelObjects.js
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsIntention.js]:
```suggestion
 *
 */
```

File: leaf-ui/rappid-extensions/BackboneModelsIntention.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        repStop: null,          // B, C, D, ..
+        repCount: null,         // 2, 3, ..n
+        repAbsTime: null,       // 0, 1, ..n
+       
+    },
+    
+    //TODO: I think could have some helper functions for the constructor. 
+});
+
+/** This file contains backbone model representations of the original model objects - WIP */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsIntention.js]: Can remove

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+ *  Actor
+ *  Link
+ *  UserEvaluation
+ *  Constraint
+ */
+
+
+/** 
+ * Backbone Model of Actor
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js]:
```suggestion
 * 
 */
```

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    initialize: function(options){ 
+        this.type = options.type;       // Options are 'Actor', 'Agent', and 'Role'
+        this.name = options.name;
+    }
+});
+
+
+/** 
+ * Backbone Model of Link.
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js]:
```suggestion
 * 
 */
```

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        linkType:  'AND',
+        postType: null,
+        absTime: -1
+    }
+});
+
+
+/** 
+ * Backbone Model of UserEvaluations 
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js]:
```suggestion
 * 
 */ 
```

File: leaf-ui/rappid-extensions/BackboneModelsOther.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        this.intentionID = options.intentionID; 
+        // TODO: Is absTP a letter or a number? It may need to be renamed.
+        this.absTP = options.absTP;             
+        this.assignedEvidencePair = options.assignedEvidencePair; //Evidence Pair
+    }
+});
+
+/** 
+ * Backbone Model of Constraints
+ * */
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/BackboneModelsOther.js]:
```suggestion
 * 
 */
```

**Review by leshghi — APPROVED** [review]
Looks good to me as base classes

File: leaf-ui/rappid-extensions/BackboneModelsOther.js (hunk 2 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/BackboneModelsOther.js:63]: Let's leave for now.

**Pull request merged**